client/src/components/Gallery.jsx

```diff
@@ -1,98 +1,80 @@
 import React from 'react';
 import styled from 'styled-components';
 
+import Modal from 'react-bootstrap/Modal';
+import ModalBody from 'react-bootstrap/ModalBody';
+import ModalDialog from 'react-bootstrap/ModalDialog';
+
 import { Container, Row, Col, Image } from 'react-bootstrap'
 
+import Carousel from './Carousel.jsx';
+
+
 class Gallery extends React.Component {
   constructor(props) {
     super(props)
     this.state = {
-
+      carModalIsOpen: false,
+      caroImage: ''
     }
   }
 
-  // layout(images, min, max) {
-  //   let result = [[images[0]]];
-  //   let random;
-  //   let i = 1;
-
-  //   while (i < images.length) {
-  //     random = Math.floor(Math.random() * (max - min)) + min;
-  //     result.push(images.slice(i, i + random));
-  //     i += random;
-  //   }
-
-  //   if (result.some(j => j.length < min)) {
-  //     return layout(images, min, max);
-  //   } else {
-  //     return result;
-  //   }
-  // }
-
+  handleClick(e) {
+    var image = e.target.src;
+    this.setState({
+      caroImage: image
+    }, () => {
+      this.toggleModal();
+    })
+  }
 
+  toggleModal() {
+    this.setState({
+      carModalIsOpen: !this.state.carModalIsOpen
+    });
+  }
 
 
   render() {
     return (
-      <Container fluid style={{paddingLeft: 0, paddingRight: 0}}>
-        {this.props.images.map(imageRow => (
-          <Row>
-            {imageRow.map(imageCol => (
-                <Col style={{paddingLeft: 0, paddingRight: 0}}>
-                  <Picture>
-                    <img src={imageCol} className="gallery-image" />
-                  </Picture>
-                </Col>
-              ))
-            }
-          </Row>
-        ))}
-      </Container>
+      <React.Fragment>
+        <Container fluid style={{paddingLeft: 0, paddingRight: 0}}>

```

Comment:
What is fluid?

Reply:
This is a bootstrap thing. Sort of like a flexbox thing for bootstrap elements.